test/token/ERC20WithPermit.test.js

```diff
@@ -114,8 +134,57 @@ describe("ERC20WithPermit", () => {
         })
       })
 
+      context("when the sender transfers part of their tokens", () => {
+        const amount = to1e18(60)
+
+        it("should transfer the requested amount", async () => {
+          await token.connect(initialHolder).transfer(recipient.address, amount)
+
+          expect(await token.balanceOf(initialHolder.address)).to.equal(
+            initialHolderBalance.sub(amount)
+          )
+
+          expect(await token.balanceOf(recipient.address)).to.equal(amount)
+        })
+
+        it("should emit a transfer event", async () => {
+          const tx = await token
+            .connect(initialHolder)
+            .transfer(recipient.address, amount)
+
+          await expect(tx)
+            .to.emit(token, "Transfer")
+            .withArgs(initialHolder.address, recipient.address, amount)
+        })
+      })
```

Comment:
Consider calling `transfer` just once in the whole context (for example in `beforeEach`):
```
context("when the sender transfers part of their tokens", () => {
  const amount = to1e18(60)
  let tx
  beforeEach(async () => {
    tx = await token
      .connect(initialHolder)
      .transfer(recipient.address, amount)
  })

  it("should transfer the requested amount", async () => {
    expect(await token.balanceOf(initialHolder.address)).to.equal(
      initialHolderBalance.sub(amount)
    )

    expect(await token.balanceOf(recipient.address)).to.equal(amount)
  })

  it("should emit a transfer event", async () => {
    await expect(tx)
      .to.emit(token, "Transfer")
      .withArgs(initialHolder.address, recipient.address, amount)
  })
})
```

I saw this pattern in most unit tests, for example for [`mint` tests](https://github.com/thesis/solidity-contracts/blob/431fb388b9aee13869ebf69d6967d1e89b64cd19/test/token/ERC20WithPermit.test.js#L699).

Reply:
0ef7ee3cc8758b9613e107625494da8b4d4ce51f